src/components/WorkerPoolCreateForm.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <p-form class="worker-pool-create-form" @submit="submit">
+    <p-content>
+      <p-label label="Name " :message="nameErrorMessage" :state="nameState">
+        <p-text-input v-model="name" :state="nameState" />
+      </p-label>
+
+      <p-label label="Description (Optional)">
+        <p-textarea v-model="description" rows="7" />
+      </p-label>
+
+      <p-label label="Type" :state="typeState" :message="typeErrorMessage">
+        <WorkerPoolTypeSelect v-model:selected="type" :state="typeState" />
+      </p-label>
+
+      <p-label label="Status (Optional)">
+        <p-toggle v-model="isActive">
+          <template #append>
+            <div>
+              <template v-if="isActive">
+                Active
+              </template>
+              <template v-else>
+                Paused
+              </template>
+            </div>
+          </template>
+        </p-toggle>
+      </p-label>
+
+      <p-label label="Flow Run Concurrency (Optional)">
+        <p-number-input v-model="concurrencyLimit" placeholder="Unlimited" :min="0" />
+      </p-label>
+    </p-content>
+
+    <template #footer>
+      <p-button inset @click="cancel">
+        Cancel
+      </p-button>
+      <SubmitButton action="Create" :loading="pending" />
+    </template>
+  </p-form>
+</template>
+
+<script lang="ts" setup>
+  import { showToast } from '@prefecthq/prefect-design'
+  import { useValidation, useValidationObserver, ValidationRule } from '@prefecthq/vue-compositions'
+  import { computed, ref } from 'vue'
+  import { useRouter } from 'vue-router'
+  import { SubmitButton, WorkerPoolTypeSelect } from '@/components'
+  import { useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
+  import { localization } from '@/localization'
+
+  const api = useWorkspaceApi()
+  const router = useRouter()
+  const routes = useWorkspaceRoutes()
+
+  const { validate, pending } = useValidationObserver()
+
+  const name = ref('')
+  const description = ref()
+  const type = ref('')
+  const concurrencyLimit = ref()
+  const isPaused = ref()
```

Comment:
why do some have default `''` and other `undefined`?

Reply:
also, the ones that are just `ref()` will be type `Ref<any>`. can you narrow those? `const description = ref<string>()`

Reply:
`''` needed for validation to work when we click Create button without touching any fields

Reply:
if the value was `undefined` it would pass validation?

Reply:
It was like that, yes, but once I added types to refs just now I get another error. If I don't have `''` in  `const name = ref<string>()`, I get this

<img width="1512" alt="Screenshot 2022-12-21 at 4 22 06 PM" src="https://user-images.githubusercontent.com/40467112/209005201-8738999f-3270-444c-9d05-cbdc06741333.png">

Reply:
those errors are accurate, the only reason you didn't see them before is because the underlying type `any` was swallowing errors. Theoretically you could have assigned a rule that only accepts `Date` for example. Those rules should probably accept `string | undefined` since that's what the values actually are

Reply:
I'm not sure what change I need to make here?